semantic_conventions/resource/container.yaml

```diff
@@ -32,3 +32,31 @@ groups:
         brief: >
           Container image tag.
         examples: ['0.1']
+      - id: image.id
+        type: string
+        brief: >
+          Runtime specific image identifier. Usually a hash algorithm followed by a UUID.
+        note:
+          On docker, `container.image.id` corresponds to the `Image` field from the Docker container inspect [API](https://docs.docker.com/engine/api/v1.43/#tag/Container/operation/ContainerInspect) endpoint.
+        examples: ['sha256:19c92d0a00d1b66d897bceaa7319bee0dd38a10a851c60bcec9474aa3f01e50f']
+      - id: command
+        type: string
+        requirement_level:
+          conditionally_required: If using embedded credentials or sensitive data, it is recommended to remove them to prevent potential leakage.
```

Comment:
I wonder if we shouldn't use the same guidelines added "recently" for the database semconv https://github.com/open-telemetry/semantic-conventions/blob/main/semantic_conventions/trace/database.yaml#L211

I feel the "conditionally_required" here is lacking actually a condition. It's just a recommendation to me.

Reply:
Maybe these should be made Opt-in? Then I feel it's more on the user to know if they are providing sensitive data they don't add it? 🤔

Reply:
I followed `conditionally_required` approach as per https://github.com/open-telemetry/semantic-conventions/pull/39#discussion_r1203180508

Reply:
Understood. I'm leaving it "as-in" for this PR to address it in subsequent issues/PRs unless you tell me otherwise.

Reply:
Not sure I see much point in merging this new field and right after discussing this again, but I don't want to block this over this concern. If others agree then I'm fine with it 👍 

If you look at the process [semantic convention file](https://github.com/open-telemetry/semantic-conventions/blob/main/specification/resource/semantic_conventions/process.md?plain=1#L39), the "additional requirements" only actually applies to `process.command_args`  and `command_line`.. so seems already "wrong" there 🤔

Reply:
To me it seems they would be better off by being Opt-in, like @jsuereth mentioned initially https://github.com/open-telemetry/semantic-conventions/pull/39#discussion_r1202319562. Maybe we also have to make the process ones opt in as well, given the recommendation there seems a bit off.

Reply:
Updated them to be `opt-in` in 400f8214004d35349b2f55c8717de1cac12b2d93.